cg/store/crud/read.py

```diff
@@ -1545,3 +1542,61 @@ def get_case_ids_for_samples(self, sample_ids: list[int]) -> list[str]:
         for sample_id in sample_ids:
             case_ids.extend(self.get_case_ids_with_sample(sample_id))
         return list(set(case_ids))
+
+    def get_related_dna_samples_from_sample_within_collaborators(
+        self, sample_internal_id: str, collaborators: set[Customer]
+    ) -> list[Sample]:
+        """Returns a DNA sample with the same subject_id, tumour status and within the collaborators of an RNA sample."""
+        sample: Sample = self.get_sample_by_internal_id(internal_id=sample_internal_id)
+
+        sample_application_version_query: Query = self._get_join_sample_application_version_query()
+        dna_prep_categories: list[PrepCategory] = [
+            PrepCategory.WHOLE_GENOME_SEQUENCING,
+            PrepCategory.TARGETED_GENOME_SEQUENCING,
+            PrepCategory.WHOLE_EXOME_SEQUENCING,
+        ]
```

Comment:
This is still business logic that should be provided to the function instead of existing in it.

Reply:
That was my question! Moved it!